File: ruby/bonAppetit.rb
# frozen_string_literal: true

def bonAppetit(k, bill, b)
  total_bill = bill.inject(:+)
  anaOwes = (total_bill - bill[k]) / 2
  return (total_bill / 2) - anaOwes if anaOwes != b

  'Bon Appetit'
end

puts bonAppetit(1, [3, 10, 2, 9], 7)

File: javascript/bonAppetit.js
let bill = [4, 1];
let k = [3, 10, 2, 9]
let b = 7

function bonAppetit(bill, k, b) {

    let anaOwes = ((k.reduce((a, b) => a + b)) - k[1]) / 2;
    console.log(anaOwes);
    if (anaOwes !== b) {
        console.log((k.reduce((a, b) => a + b) / 2) - anaOwes)
    } else {
        console.log('Bon Appetit');
    }

}

bonAppetit(bill, k, b);

